modules/gloo/controller/service/handler.go

```diff
@@ -23,6 +23,8 @@ func Register(ctx context.Context, rContext *types.Context) error {
 		rContext.Rio.Rio().V1().Service(),
 		rContext.Apply.WithCacheTypes(
 			rContext.Gateway.Gateway().V1().VirtualService(),
+			// TODO decide if I need this
+			rContext.Core.Core().V1().Endpoints(),
```

Comment:
It works fine without it... I thought it may be needed since there is a requirement on looking up `corev1.Endpoints` but I do it with a direct API call. @StrongMonkey may need to look at this